### ISSUE TITLE
ActiveForm::validate could use a $clearErrors option

### ISSUE THREAD

**Issue by dovestyle** [yiisoft/yii2#15648]
`ActiveForm::validate()` calls `ActiveRecord::validate()` on the model, but doesn't let us set the `$clearErrors` attribute that `ActiveRercord::validate()` accepts. I think you see where this is going, but.. if we manually add any errors to the model before we call `ActiveForm::validate()`, they get wiped:

```php
// Assuming you have a User model with a name attribute..

$model = \app\models\User::findOne(1);
$model->addError('name', "I don't like your name.");
ActiveForm::validate($model);
print_r($model->getErrors());

// ..the resulting array will not include that error
```

So.. just wondering if we could get a similar attribute on the AF::validate method? I'm writing my own for now, would just prefer to use the lib's.

Cheers!

**Comment by samdark** [comment]
Would you please clarify your use case? Why can't you add error after calling `validate()`?

**Comment by dovestyle** [comment]
Wow, sorry about the minimal details, was the end of a long day yesterday.

Due to certain restrictions, we need to make `addError` calls in the controller. We are using `return ActiveForm::validate($model);` in the controller to return the validation results to the UI with the correct HTML field names. If we did `$errors = ActiveForm::validate($model);` and then added errors to the model, that would do nothing to `$errors` and doing the AF::validate call again would just wipe those errors we just added.

Even the yii2 cookbook kinda rolled their own solution, on the bottom in actionUpdate - https://yii2-cookbook.readthedocs.io/forms-activeform-js/

And we rolled our own as well:
```php
  protected function validationFields($model)
  {
    $model->validate(null, false);
    $result = [];
    foreach ($model->getErrors() as $attribute => $errors) {
      $result[Html::getInputId($model, $attribute)] = $errors;
    }
    return $result;
  }
```

The only real difference between that and AF::validate is that this passes false to `ActiveRecord::validate`'s `$clearErrors` so we're not overwriting any errors added before it's called. 

Is this enough details?

**Comment by FabrizioCaldarelli** [comment]
I apply ActiveForm::validate() only in ajaxValidation, hardly ever. In most cases, I fill the model from $_POST and then call ActiveRecord::validate();

In what other cases do you use ActiveForm::validate() ?

**Comment by dovestyle** [comment]
@FabrizioCaldarelli same use-case, except we're adding custom errors outside the model, (not part of its validation.)

**Comment by FabrizioCaldarelli** [comment]
Would it more appropriate to apply all validations in model? So would be an unique central point where validations are done

**Comment by dovestyle** [comment]
@FabrizioCaldarelli yes of course, but like I said, we have certain restrictions that disallow us from doing that in this particular task.
